### PR TITLE
feat: `ExprNameNamespace`

### LINKED ISSUE #130: Add `.name.suffix`
From a request  [here](https://github.com/koaning/scikit-lego/pull/661#discussion_r1596527264)

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Closes #130 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

## Description

Introduces [name namespace](https://docs.pola.rs/api/python/stable/reference/expressions/name.html) for both pandas-like and pyarrow. Methods not implemented refer to polars structs

### CODE REVIEW

File: tests/expr/name/map_test.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    result = df.select((nw.col("foo", "BAR") * 2).name.map(function=map_func))
+    expected = {map_func(k): [e * 2 for e in v] for k, v in data.items()}
+    compare_dicts(result, expected)
+
+
+def test_map_after_alias(constructor_with_lazy: Any) -> None:
+    df = nw.from_native(constructor_with_lazy(data))
+    result = df.select((nw.col("foo")).alias("alias_for_foo").name.map(function=map_func))
+    expected = {map_func("foo"): data["foo"]}
+    compare_dicts(result, expected)
```
Comment (FBruzzesi) [inline tests/expr/name/map_test.py:30]: These tests `<>_after_alias` are to make sure to account for what's been discussed in #186 [comment](https://github.com/narwhals-dev/narwhals/pull/186#issuecomment-2123146302)

**Comment by FBruzzesi** [comment]
As requested in the community call:

```py
import narwhals as nw
import pandas as pd

data = {"foo": [1, 2, 3], "BAR": [4, 5, 6]}
df = nw.from_native(pd.DataFrame(data))

nw.to_native(
    df.select(
        nw.sum_horizontal("foo", "BAR").alias("new_name").name.suffix("_sum_h")
   )
)
```
```bash
   foo_sum_h
0          5
1          7
2          9
```

**Comment by MarcoGorelli** [comment]
wow, seriously well done here!

could you fetch and merge upstream/main please? I think that due to the `_polars` refactor, a little change may need adding there too - but it should be quite minimal

**Review by MarcoGorelli — APPROVED** [review]
awesome, well done for sticking with this!

**Pull request merged**